invenio_app_rdm/theme/views/records.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019-2020 CERN.
+# Copyright (C) 2019-2020 Northwestern University.
+# Copyright (C)      2021 TU Wien.
+#
+# Invenio App RDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Routes for record-related pages provided by Invenio-App-RDM."""
+
+from flask import abort, g, render_template, request
+from invenio_files_rest.views import ObjectResource
+from invenio_rdm_records.proxies import current_rdm_records
+
+from ..utils import obj_or_import_string, previewer_record_file_factory
+
+
+def register_records_ui_routes(app, blueprint):
+    """Dynamically registers routes (allows us to rely on config)."""
+
+    @blueprint.route(
+        app.config.get(
+            "APP_RDM_RECORDS_EXPORT_URL",
+            "/records/<pid_value>/export/<export_format>",
+        )
+    )
+    def export_record(pid_value, export_format="json", **kwargs):
+        """Record export-as landing page."""
+        template = kwargs.get(
+            "template", "invenio_app_rdm/landing_page/export_page.html"
+        )
+
+        resource = current_rdm_records.records_resource
+        service = current_rdm_records.records_service
+        links_config = resource.config.links_config
+
+        record = service.read(
+            id_=pid_value, identity=g.identity, links_config=links_config
+        )._record
+
+        # get the configured serializer
+        temp = app.config.get("APP_RDM_RECORD_EXPORTERS", {}).get(
+            export_format
+        )
+        if temp is None:
+            raise Exception(
+                "no exporter for the specified format registered: {}".format(
+                    export_format
+                )
+            )
+
+        format_name = temp.get("name", export_format)
+        serializer = obj_or_import_string(temp["serializer"])()
```

Comment:
Yeah... UIJSONSerializer isn't currently implemented to allow the usage of the pretty_print options we talked about :disappointed: ... I made an issue outlining the problem and solution here: https://github.com/inveniosoftware/invenio-rdm-records/issues/367 . If you want to take it separately, it would help :+1: .